core/dbt/adapters/materialization/README.md

```diff
@@ -0,0 +1,111 @@
+In our scenarios, `context` tends to be either an instance of `RelationConfig` corresponding to the new state
+or a single value if the change is simple. For example, creating an index would require the entire config;
+whereas updating a setting like autorefresh for Redshift would require only the setting.
+
+### RelationConfigChangeset
+This class is effectively a bin for collecting instances of `RelationConfigChange`. It comes with a few helper
+methods that facilitate rolling up concepts like `require_full_refresh` to the aggregate level.
+
+### RelationConfigValidationMixin
+This mixin provides optional validation mechanics that can be applied to either `RelationConfigBase` or
+`RelationConfigChange` subclasses. A validation rule is a combination of a `validation_check`, something
+that should evaluate to `True`, and an optional `validation_error`, an instance of `DbtRuntimeError`
+that should be raised in the event the `validation_check` fails. While optional, it's recommended that
+the `validation_error` be provided for clearer transparency to the end user.
+
+## Basic Building Blocks (Data)Classes
+
+### DatabaseConfig
+This is the most basic version of `RelationConfig` that we can have. It adds a `name` and a `fully_qualified_path`
+and nothing else. But we generally need a database when dbt runs. In particular, we need to reference a database
+in other classes, like `SchemaConfig`.
+
+### SchemaConfig
+As with `DatabaseConfig`, this class is pretty basic. It's existence is the most important thing. While this
+may not be needed for certain databases, it's prevalent enough that it's worth building it as an out-of-the-box
+object.
+
+### MaterializationConfig
+This is the pearl in the sand. dbt generally interacts at the materialization level. As an adapter maintainer, you'll
+need to subclass from most, if not all, objects in `materialization_config`; however you're likely doing so in order
+to work with a `MaterializationConfig` subclass. This class has several helper methods that make it easier
+to template sql in jinja.
+
+### IncludePolicy
+Identifies whether a component is included in a fully qualified path.
+
+### QuotePolicy
+Identifies whether a component is considered case-sensitive, or should be quoted, in a fully qualified path. This
+config also contains the quote character for the database.
+
+## Functions
+
+### Policy
+There are only a handful of functions here. In order to reduce duplication in docs, the reader is referred to the
+docstrings on these functions for more detail:
+
+- `conform_part`
+- `render_part`
+- `render`
+
+### Materialization
+This is a new class that serves as a service layer to expose `MaterializationConfig` functionality in the
+jinja context. We're effectively tucking away the modelling of database objects in python and only exposing
+class methods to serve as a basic API into `MaterializationConfig`.
```

Comment:
Currently dbt classes available in the jinja context are documented here: https://docs.getdbt.com/reference/dbt-classes. Let's open an issue to do the same for `Materialization` in https://github.com/dbt-labs/docs.getdbt.com/. I could see an argument in holding off on it until the pattern is adopted more broadly for other materialization types though.

Reply:
I added this to my backlog, associated with the MV epic.